src/Microsoft.Identity.Client/WsTrust/WsTrustWebRequestManager.cs

```diff
@@ -139,12 +139,33 @@ public async Task<MexDocument> GetMexDocumentAsync(string federationMetadataUrl,
         {
             requestContext.Logger.Info("Sending request to userrealm endpoint.");
 
-            var uri = new UriBuilder(userRealmUriPrefix + userName + "?api-version=1.0").Uri;
+            UriBuilder builder = CreateUserrealmEndpoint(userRealmUriPrefix, userName, requestContext);
 
-            var httpResponse = await _httpManager.SendGetAsync(uri, null, requestContext).ConfigureAwait(false);
+            var httpResponse = await _httpManager.SendGetAsync(builder.Uri, null, requestContext).ConfigureAwait(false);
             return httpResponse.StatusCode == System.Net.HttpStatusCode.OK 
                 ? JsonHelper.DeserializeFromJson<UserRealmDiscoveryResponse>(httpResponse.Body) 
                 : null;
         }
-    }
+
+        public UriBuilder CreateUserrealmEndpoint(
```

Comment:
If you don't need the UriBuilder to make more changes to this uri, just return the Uri.

Reply:
This would be even better and if you do this (just return the Uri) ignore my other comment about renaming the CreateUserRealmEndpoint method.

---
In reply to: [251520633](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/840#discussion_r251520633) [](ancestors = 251520633)